subsys/net/ip/connection.c

```diff
@@ -211,6 +211,12 @@ static struct net_conn *conn_find_handler(uint16_t proto, uint8_t family,
 				continue;
 			}
 
+#if defined(CONFIG_NET_CONTEXT_REUSEADDR)
+			if (conn->context->options.reuseaddr != 0) {
+				continue;
+			}
+#endif
```

Comment:
To me this looks like we are doing this too early. We are now ignoring all the checks done below. Better would be to check the reuseaddr if we have found a duplicate entry.

Reply:
You are definitely right. I moved to the end after all other checks in my last commit